hcipy/fourier/fast_fourier_transform.py

```diff
@@ -39,10 +45,66 @@ def make_fft_grid(input_grid, q=1, fov=1):
 
 	delta = (2 * np.pi / (input_grid.delta * input_grid.dims)) / q
 	dims = (input_grid.dims * fov * q).astype('int')
-	zero = delta * (-dims / 2 + np.mod(dims, 2) * 0.5)
+	zero = delta * (-dims / 2 + np.mod(dims, 2) * 0.5) + shift
 
 	return CartesianGrid(RegularCoords(delta, dims, zero))
 
+def reconstruct_fft_grid_parameters(input_grid, fft_grid):
```

Comment:
Does it make sense to name the grid to check fft_grid? Because its not necessarily an FFT grid. Its just a grid that is tested.

Reply:
`check_fft_grid()` feels to me like it should return either `True` or `False`, which is not what this function is doing / is supposed to be doing. But I strongly dislike the current name too. I'm open to suggestions. I tried `recover_fft_parameters()`, `reconstruct_fft_parameters()`, and `get_fft_grid_parameters()` and some variants. I don't like either of them.

Reply:
I think the function name is fine, but the parameter name fft_grid is a little bit confusing. I thought I would need to supply an fft grid to get the best matching fft grid parameters. Which is sort of true, because you only return thing if it actually is an fft grid. But it got me confused about what the function is trying to do.

Maybe the function should be split into two? A check if some grid is an FFT grid, and a recover parameters function? Because you now rely on the raising of ValueErrors for other functions. And I am not sure if that is the proper way of using warnings and errors.

Reply:
Gotcha, I misread that. The function I wrote assumes that `fft_grid` is a native FFT grid, and raises exceptions when this is not the case. Python uses exceptions way more liberal than you'd think (`IndexError`, `AttributeError`, `StopIteration`, etc..). Exceptions are not _that_ exceptional, compared to how they are supposed to be used in other languages.

Another reason for doing it this way is that while doing a check of whether a grid is a native FFT grid, you essentially have to do the exact same calculations as fully reconstructing the parameters (with the exception of `shift`, a single line of code).

Maybe add the following function?
```python
def is_fft_grid(input_grid, grid): # Maybe switch the arguments around? The same for the reconstruct... function?
    try:
        reconstruct_fft_grid_parameters(input_grid, grid)
    except ValueError:
        return False
    return True
```

The point is: `reconstruct_fft_grid_parameters()` (or whatever we end up calling it) will _need_ to raise an exception if the input grid is not a native FFT grid, as it has no way to signify this to the caller in any other way. Correct me if I'm wrong on that.

Reply:
I think that is correct. You can't recover the parameters if the grid is not an fft grid. So raising a Value Error is the correct approach. I like the second function. I know it will add some overhead, but it's clearer what to expect back and how to check for an fft_grid.